src/middleware/json-api/_serialize.js

```diff
@@ -29,7 +29,14 @@ function resource (modelName, item) {
   })
 
   serializedResource.type = typeName
-  serializedResource.attributes = serializedAttributes
+
+  var attrValues = Object.keys(serializedAttributes).map(key => {
+    return serializedAttributes[key]
+  })
+
+  if (!!attrValues && attrValues.filter(val => val === undefined).length !== attrValues.length) {
```

Comment:
don't include `attributes` key if all attributes are `undefined`

Reply:
It might be safer and more intention-revealing to do `Boolean(attrValues)` here instead of using `!!`.

Reply:
👍